src/v2/view-builder/views/email/ChallengeAuthenticatorEmailView.js

```diff
@@ -0,0 +1,119 @@
+import { addSwitchAuthenticatorLink } from '../../utils/AuthenticatorUtil';
+import { SHOW_RESEND_TIMEOUT } from '../../utils/Constants';
+
+const ResendView = View.extend(
+  {
+    className: 'hide resend-email-view',
+    events: {
+      'click a.resend-link' : 'handelResendLink'
+    },
+
+    initialize () {
+      this.add(createCallout({
+        content: `${loc('email.code.not.received', 'login')}
+        <a class='resend-link'>${loc('email.button.resend', 'login')}</a>`,
+        type: 'warning',
+      }));
+    },
+
+    handelResendLink () {
+      this.options.appState.trigger('invokeAction', 'currentAuthenticatorEnrollment-resend');
+      // Hide warning, but reinitiate to show warning again after some threshold of polling
+      if (!this.el.classList.contains('hide')) {
+        this.el.classList.add('hide');
+      }
+      this.showCalloutWithDelay();
+    },
+
+    postRender () {
+      this.showCalloutWithDelay();
+    },
+
+    showCalloutWithDelay () {
+      this.showMeTimeout = _.delay(() => {
+        this.$el.removeClass('hide');
+      }, SHOW_RESEND_TIMEOUT);
+    },
+
+    remove () {
+      View.prototype.remove.apply(this, arguments);
+      clearTimeout(this.showMeTimeout);
+    }
+  },
+);
+
+const Body = BaseForm.extend(Object.assign(
+  {
+    save () {
+      return loc('oie.verify.button', 'login');
+    },
+    initialize () {
+      BaseForm.prototype.initialize.apply(this, arguments);
+      const sendText = loc('oie.email.verify.sentText', 'login');
+      const enterCodeText = loc('oie.email.verify.codeText', 'login');
+
+      // Courage doesn't support HTML, hence creating a subtitle here.
+      this.add(`<div class="okta-form-subtitle" data-se="o-form-explain">
+        ${sendText}&nbsp;<span class='strong'>${this.model.escape('email')}.</span>
+        ${enterCodeText}</div>`, {
+        prepend: true,
+        selector: '.o-form-error-container',
+      });
+      // polling has been killed when click save to avoid race conditions hence resume if save failed.
+      this.listenTo(this.options.model, 'error', this.startPolling.bind(this));
+    },
+
+    saveForm () {
+      BaseForm.prototype.saveForm.apply(this, arguments);
+      this.stopPolling();
+      // TODO: abort ongoing request. (https://oktainc.atlassian.net/browse/OKTA-244134)
+    },
+
+    postRender () {
+
+      this.add(ResendView, {
+        selector: '.o-form-error-container',
+      });
+      this.startPolling();
+    },
+
+    remove () {
+      BaseForm.prototype.remove.apply(this, arguments);
+      this.stopPolling();
+    }
+  },
+
+  email,
+  polling,
+));
+
+const Footer = BaseFooter.extend({
+  links () {
+    const links = [];
+    addSwitchAuthenticatorLink(this.options.appState, links);
+    return links;
+  }
+});
+
+export default BaseFactorView.extend({
+  Body,
+  Footer,
+  createModelClass () {
+    const { profile } = this.options.appState.get('currentAuthenticatorEnrollment');
```

Comment:
`this.options.currentViewState.relatesTo.value`

Reply:
There seems to be no `relatesTo` in `this.options.currentViewState`, it's also not there in the mock..

Reply:
I think we can update mock.

Reply:
I updated it with one from my IDX org (from postman).